Scripts/Runtime/Data/Collections/UnsafeHeap.cs

```diff
@@ -0,0 +1,610 @@
+using Anvil.Unity.DOTS.Entities;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using Unity.Collections;
+using Unity.Collections.LowLevel.Unsafe;
+
+namespace Anvil.Unity.DOTS.Data
+{
+    //NOTE - Taken from: https://github.com/Amarcolina/NativeHeap (MIT LICENCE)
+    //This is a lightly cleaned up version of NativeHeap from Amarcolina that removes
+    //any safety aspects so that it can be used inside a different Native Collection.
+    
+    /// <summary>
+    /// This is a basic implementation of the MinHeap/MaxHeap data structure.  It allows you
+    /// to insert objects into the container with a O(log(n)) cost per item, and it allows you
+    /// to extract the min/max from the container with a O(log(n)) cost per item.
+    /// 
+    /// This implementation provides the ability to remove items from the middle of the container
+    /// as well.  This is a critical operation when implementing algorithms like astar.  When an
+    /// item is added to the container, an index is returned which can be used to later remove
+    /// the item no matter where it is in the heap, for the same cost of removing it if it was
+    /// popped normally.
+    /// 
+    /// This container is parameterized with a comparer type that defines the ordering of the
+    /// container.  The default form of the comparer can be used, or you can specify your own.
+    /// The item that comes first in the ordering is the one that will be returned by the Pop
+    /// operation.  This allows you to use the comparer to parameterize this collection into a 
+    /// MinHeap, MaxHeap, or other type of ordered heap using your own custom type.
+    /// 
+    /// For convenience, this library contains the Min and Max comparer, which provide
+    /// comparisons for all built in primitives.
+    /// </summary>
```

Comment:
Based on discussion can we add a `<remarks>` section that mentions this type does not have the safety that other unsafe types have in anvil?

Also a TODO mentioning that this will be replaced by the work in (some issue number) so fixing the above remark isn't worthwhile.

Reply:
Done